linode/manager
# M3-5988: Add SMTP restriction support ticket form

## Description 📝

**What does this PR do?**

Adds form fields specific to lifting SMTP restrictions to a support ticket. This is part of efforts to improve the user experience and reduce the amount of back-and-forth between customers and Support when handling requests to remove these restrictions.

**Notes**: 

- You may notice when testing various types of support tickets that the title and description fields are prepopulated with data that does not belong to that support ticket, but was from a previous support ticket, likely because form values are cached in localstorage and not cleared. This issue seems to exist in production prior to these changes and is beyond the scope of this ticket, but could be addressed in another.
- Will create an additional ticket to transition `SupportTicketDrawer` to Formik for improved form state management.


## Preview 📷
**Trigger and fill out the support ticket:**

https://user-images.githubusercontent.com/114685994/206579933-9f1caa2f-7eac-482f-8d5c-4cabe9b85c33.mov


**Submitted support ticket:**
![Screen Shot 2022-12-08 at 2 22 00 PM](https://user-images.githubusercontent.com/114685994/206747496-b5f1240d-c11b-4975-ac2e-5a0c40fdcd38.jpg)


## How to test 🧪

**What are the steps to reproduce the issue or verify the changes?**
1. **Use the dev environment for the following to avoid generating test support tickets in production:**
2. Use an account that has been created since _or temporarily modified_ to have an account creation date of on/after 11-30-22 (the date we activated the SMTP restriction notices for new accounts).
3. Navigate to the linode details page for a linode and select the network tab.
4. Click on the "open a support ticket" link in the SMTP restriction notice.
5. Observe that you've been redirected to the Support page with an open support dialog that has form fields specific to SMTP.
6. Observe that the form does not submit until a user enters all required fields. 
7. Fill out all required fields and submit the ticket. 
8. Observe that the user is redirected to the ticket details page. Check that ticket details look correct and readable.
9. Check that Support can view the ticket and that details look correct and readable.
10. Confirm that a user can access the form dialog for a general support ticket and a support ticket with entities, and the SMTP form does _not_ display. (Turning on MSW and navigating to the Volumes Landing page to click the volume with "Contact Support" in its status is a quick way.)

**How do I run relevant unit or e2e tests?**

Units tests:
```
yarn test SMTPRestrictionText
```
```
yarn test Support
```

e2e:
```
yarn cy:run -s "cypress/e2e/helpAndSupport/open-support-ticket.spec.ts"
```
